x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/calculations/overall_max.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { FormattedIndexPatternColumn, ReferenceBasedIndexPatternColumn } from '../column_types';
+import { optionallHistogramBasedOperationToExpression } from './utils';
+import { OperationDefinition } from '..';
+import { getFormatFromPreviousColumn } from '../helpers';
+
+const ofName = (name?: string) => {
+  return i18n.translate('xpack.lens.indexPattern.overallMaxOf', {
+    defaultMessage: 'Overall max of {name}',
+    values: {
+      name:
+        name ??
+        i18n.translate('xpack.lens.indexPattern.incompleteOperation', {
+          defaultMessage: '(incomplete)',
+        }),
+    },
+  });
+};
+
+export type OverallMaxIndexPatternColumn = FormattedIndexPatternColumn &
+  ReferenceBasedIndexPatternColumn & {
+    operationType: 'overall_max';
+  };
+
+export const overallMaxOperation: OperationDefinition<
```

Comment:
Is there such a big difference between each overall function that they need to be in separate files? It looks to me like the main difference is the text, but we could generate these functions the same as we generate the simple metrics?